src/lib.rs

```diff
@@ -189,6 +191,58 @@ impl Tokenizer {
         })
     }
 
+    /// Tokenize a batch of multiple input strings.
+    ///
+    /// Each given input string is tokenized and the numeric representation
+    /// written to a row in the resulting two-dimensional matrix of shape
+    /// `(texts.len(), context_length)`, with the special `<start_of_text>`
+    /// token prepended, and `<end_of_text>` appended to each text.
+    ///
+    /// The individual input strings are lowercased before being tokenized, but
+    /// otherwise no pre-processing is performed.
+    ///
+    /// `context_length` is the maximum number of tokens per each text and
+    /// should be `77` for all current CLIP models. If tokenizing an input text
+    /// results in too many tokens, the token sequence will be truncated to fit
+    /// within the resulting row of length `context_length`, always including
+    /// the `<start_of_text>` and `<end_of_text>` marker tokens.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use ndarray::array;
+    /// # use instant_clip_tokenizer::{Token, Tokenizer};
+    /// let tokenizer = Tokenizer::new();
+    /// let encoded = tokenizer.tokenize_batch(["Hi", "How are you?"], 5);
+    /// assert_eq!(encoded, array![
+    ///     [49406, 1883, 49407, 0, 0],
+    ///     [49406, 829, 631, 592, 49407],
+    /// ]);
+    /// ```
+    pub fn tokenize_batch<'a, I>(&self, texts: I, context_length: usize) -> Array2<u16>
```

Comment:
I prefer to use `impl IntoIterator<Item = &'a str>` over separate bounds where possible, since this saves the indirection of a type argument name.

Reply:
> I prefer to use impl IntoIterator<Item = &'a str> over separate bounds where possible, since this saves the indirection of a type argument name.

I'm not sure how this can be done, given that we also need the `I::IntoIter: ExactSizeIterator` bound?

> Also, I think we should name this to match the other methods, so probably encode_batch()?

I intentionally chose a different name because it does a bit more than `encode`, namely add `<start_of_text>` and  `<end_of_text>` markers, and truncate to `context_length`.
The names `tokenize` and `encode` are also what the original Python library uses for the corresponding functions.

Reply:
> > I prefer to use impl IntoIterator<Item = &'a str> over separate bounds where possible, since this saves the indirection of a type argument name.
> 
> I'm not sure how this can be done, given that we also need the `I::IntoIter: ExactSizeIterator` bound?

Ah, that's fair -- sorry for the noise.

> > Also, I think we should name this to match the other methods, so probably encode_batch()?
> 
> I intentionally chose a different name because it does a bit more than `encode`, namely add `<start_of_text>` and `<end_of_text>` markers, and truncate to `context_length`. The names `tokenize` and `encode` are also what the original Python library uses for the corresponding functions.

Alright. Just make sure the difference is very clear in the documentation.